pilot/pkg/bootstrap/server.go

```diff
@@ -276,6 +276,13 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if err := s.initMeshNetworks(&args); err != nil {
 		return nil, fmt.Errorf("mesh networks: %v", err)
 	}
+	// Certificate controller must be created before MCP
+	// controller to avoid deadlocks caused by the circular dependency, i.e.,
+	// MCP server pod waits to mount a certificate while
+	// Pilot waits for MCP server.
+	if err := s.initCertController(&args); err != nil {
```

Comment:
I understand the init here just create a controller, what really matters is the start func `s.certController.Run(stop)`.

Reply:
I mean in order to speed up istio bootstrap, could we create the necessary certs synchronously, and leave the reconcile in the `certController.Run` which is asynchronously started.

Reply:
The subject and the implementation of this PR has been revised: Separate DNS certificate from other certificates (e.g., SPIFFE, SDS, etc) so each certificate can be used for a specific purpose.

Reply:
Ah, noticed